xchange-kraken/src/main/java/org/knowm/xchange/kraken/KrakenAdapters.java

```diff
@@ -259,13 +259,21 @@ public static CurrencyPair adaptCurrencyPair(String krakenCurrencyPair) {
   }
 
   public static OpenOrders adaptOpenOrders(Map<String, KrakenOrder> krakenOrders) {
+    return adaptOpenOrders(krakenOrders, null);
+  }
+
+  public static OpenOrders adaptOpenOrders(Map<String, KrakenOrder> krakenOrders, CurrencyPair currencyPair) {
 
     List<LimitOrder> limitOrders = new ArrayList<>();
     for (Entry<String, KrakenOrder> krakenOrderEntry : krakenOrders.entrySet()) {
       KrakenOrder krakenOrder = krakenOrderEntry.getValue();
       KrakenOrderDescription orderDescription = krakenOrder.getOrderDescription();
 
-      if (!"limit".equals(orderDescription.getOrderType().toString())) {
+      if (!"limit".equals(orderDescription.getOrderType().toString()) ||
+              ( currencyPair != null &&
+                !currencyPair.equals(adaptCurrencyPair(orderDescription.getAssetPair()))
+              )
+      ) {
```

Comment:
please extract this check to some method with a meaningful name like `if (isNotPlainLimitOrderOr...()) {`